pkgs/development/python-modules/minichain/default.nix

```diff
@@ -0,0 +1,65 @@
+{ buildPythonPackage
+, eliot
+, fetchPypi
+, google-search-results
+, jinja2
+, lib
+, manifest-ml
+, openai
+, pytestCheckHook
+, pythonRelaxDepsHook
+}:
+
+buildPythonPackage rec {
+  pname = "minichain";
+  version = "0.3.3";
+
+  # See https://github.com/srush/MiniChain/issues/23 and https://github.com/NixOS/nixpkgs/issues/248185 as to why we
+  # don't fetchFromGitHub.
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-+mju1Mz/aojROpiOVzv6WoRNTrhgCub4yyYLEWcHWh0=";
+  };
+
+  # See https://github.com/srush/MiniChain/issues/24.
+  postPatch = ''
+    substituteInPlace ./minichain/__init__.py --replace "from .gradio import GradioConf, show" ""
+  '';
+
+  nativeBuildInputs = [ pythonRelaxDepsHook ];
+
+  pythonRemoveDeps = [
+    # Only used in the examples:
+    "datasets"
+    "faiss-cpu"
+    "jinja2-highlight"
+    "trio"
+
+    # Not used anywhere:
+    "eliot-tree"
+
+    # Not yet packaged in nixpkgs:
+    "gradio"
```

Comment:
FYI: gradio has recently been added to nixpkgs-staging in https://github.com/NixOS/nixpkgs/pull/241558.
I expect we'll be able to add it to the "optional" dependencies soon.

Reply:
Cool, didn't know that! I'll make a mental note that it can be added in a future PR